Packages/com.unity.inputsystem/InputSystem/Events/IMECompositionEvent.cs

```diff
@@ -2,38 +2,117 @@
+    /// </summary>
+    [StructLayout(LayoutKind.Explicit, Size = InputEvent.kBaseEventSize + sizeof(int))]
+    public struct IMECompositionEventVariableSize : IInputEventTypeInfo
+    {
+        // Before we had 0x494D4553 which corresponds to ImeCompositionStringInputEventData fixed size event with 64 character payload.
+        // 0x494D4543 corresponds to ImeCompositionInputEventData and is a different event which provides variable size array of characters after the event.
+        public const int Type = 0x494D4543;
+
+        [FieldOffset(0)]
+        public InputEvent baseEvent;
+
+        [FieldOffset(InputEvent.kBaseEventSize)]
+        internal int length;
+
+        internal static unsafe char* GetCharsPtr(IMECompositionEventVariableSize* ev)
+        {
+            return (char*)((byte*)ev + InputEvent.kBaseEventSize + sizeof(int));
+        }
+
+        public FourCC typeStatic => Type;
+
+        /// <summary>
+        /// Returns composition string for the given event.
+        /// </summary>
+        /// <param name="ev">Pointer to the event.</param>
+        /// <returns>Composition string.</returns>
+        public static unsafe IMECompositionString GetIMECompositionString(IMECompositionEventVariableSize* ev)
+        {
+            return new IMECompositionString(GetCharsPtr(ev), ev->length);
+        }
+
+        /// <summary>
+        /// Queues up an IME Composition Event. IME Event sizes are variable, and this simplifies the process of aligning up the Input Event information and actual IME composition string.
+        /// </summary>
+        /// <param name="deviceId">ID of the device (see <see cref="InputDevice.deviceId"/>) to which the composition event should be sent to. Should be an <see cref="ITextInputReceiver"/> device. Will trigger <see cref="ITextInputReceiver.OnIMECompositionChanged"/> call when processed.</param>
+        /// <param name="str">The IME characters to be sent. This can be any length, or left blank to represent a resetting of the IME dialog.</param>
+        /// <param name="time">The time in seconds, the event was generated at. This uses the same timeline as <see cref="Time.realtimeSinceStartup"/></param>
+        public static unsafe void QueueEvent(int deviceId, string str, double time)
+        {
+            var sizeInBytes = (InputEvent.kBaseEventSize + sizeof(int)) + sizeof(char) * str.Length;
+            var eventBuffer = new NativeArray<byte>(sizeInBytes, Allocator.Temp, NativeArrayOptions.UninitializedMemory);
+
+            var ev = (IMECompositionEventVariableSize*)eventBuffer.GetUnsafePtr();
+
+            ev->baseEvent = new InputEvent(Type, sizeInBytes, deviceId, time);
+            ev->length = str.Length;
+
+            if (str.Length > 0)
+                fixed(char* p = str)
+                UnsafeUtility.MemCpy(GetCharsPtr(ev), p, str.Length * sizeof(char));
+
+            InputSystem.QueueEvent(new InputEventPtr((InputEvent*)ev));
+
+            eventBuffer.Dispose();
```

Comment:
After a bit of digging to understand InputSystem.QueueEvent it seems it makes a copy of the input event and therefore this dispose is valid. No feedback here. Just noting for my own reminder.